packages/strapi-deepsearch-service/__tests__/component-utils.test.ts

```diff
@@ -1,22 +1,47 @@
-import { StrapiBuilder } from "strapi-builder";
-import { StrapiGlobalComponents } from "strapi-types";
+import { Strapi, StrapiGlobalComponents } from 'strapi-types';
 
 const {
   isModelComponent,
   isModelComponentSearchable,
   getComponentByModel,
-} = require("../src/services/utils/component-utils.js");
+} = require('../src/services/utils/component-utils.js');
 
-describe("Test Deep search component utils functions", () => {
-  global.strapi = new StrapiBuilder()
+class Builder {
+  strapi = new Strapi();
+
+  addComponents(components: StrapiGlobalComponents) {
+    this.strapi.components = {
+      ...this.strapi.components,
+      ...components,
+    };
+    return this;
+  }
+
+  build() {
+    return this.strapi;
+  }
+}
+
+const getStrapiBuilder = () => {
+  try {
+    const { StrapiBuilder } = require('strapi-builder');
```

Comment:
when it could fails? 🤔

Reply:
when you install this package because a builder is not published to npm

Reply:
and when `strapi-builder` will be public this implementation will use the original builder